packages/block-library/src/gallery/style.scss

```
@@ -43,22 +43,46 @@ figure.wp-block-gallery.has-nested-images {
 			width: auto;
 		}
 
+		// Position caption and scrim at the bottom.
+		&:has(figcaption)::before,
 		figcaption {
-			background: linear-gradient(0deg, rgba($color: $black, $alpha: 0.7) 0, rgba($color: $black, $alpha: 0.3) 70%, transparent);
+			position: absolute;
 			bottom: 0;
+			right: 0;
+			left: 0;
+			max-height: 100%;
+		}
+
+		// Create a background blur layer.
+		&:has(figcaption)::before {
+			content: "";
+			z-index: 0;
+			height: 100%;
+			max-height: 40%;
+
+			// Blur the background under the gradient scrim.
+			backdrop-filter: blur(4px);
+
+			// Crop the caption so the blur is only on the edge.
+			mask-image: linear-gradient(0deg, rgba($color: $black, $alpha: 1) 20%, rgba($color: $black, $alpha: 0) 100%);
+		}
+
+		// Place the caption at the bottom.
+		figcaption {
+			z-index: 1;
 			color: $white;
+			text-shadow: 0 0 1.5px rgba($color: $black, $alpha: 1);
```

```suggestion
			text-shadow: 0 0 1.5px $black;
```

Since `rgba($color: $black, $alpha: 1)` is the same as black, I think it can be made simpler.